cnctcli/actions/reports_params.py

```diff
@@ -0,0 +1,186 @@
+from cnct import R
+from interrogatio.validators import RegexValidator, RequiredValidator, Validator
+from interrogatio.core.exceptions import ValidationError
+
+import json
+
+
+class ObjectValidator(Validator):
+
+    def validate(self, value):
+        if not value or value == "":
+            return
+        try:
+            json.loads(value)
+        except ValueError:
+            raise ValidationError("Introduced data is not a valid json object")
+
+
+def required_validator(param, validators=None):
+    if not validators:
+        validators = []
+
+    if param.get('required', False):
+        validators.append(RequiredValidator(
+            message="Please select at least one value"
+        ))
+    return validators
+
+
+def single_line(param):
+
+    return {
+        'name': param['id'],
+        'type': 'input',
+        'question_mark': '',
+        'message': f'{param["description"]}',
+        'validators': required_validator(param),
+    }
+
+
+def object_param(param):
+    validators = required_validator(param)
+    validators.append(ObjectValidator())
+
+    return {
+        'name': param['id'],
+        'type': 'input',
+        'multiline': True,
+        'default': "{}",
+        'question_mark': '',
+        'message': f'{param["description"]}',
+        'validators': validators,
+    }
+
+
+def date_range(param):
+
+    date_validator = [RegexValidator('\\d{4}-\\d{2}-\\d{2}', 'Introduced date is invalid')]
```

Comment:
May be a we can add a better validation logic to validate date. Currently value like '2020-99-99' will also be considered as valid date.

Reply:
this is temporal, library will provide a new widget type soon